### DISCUSSION TITLE
How to control positioning on a column in a snippet?

### DISCUSSION THREAD

**Discussion by FALLAI-Denis** [IBM/zopeneditor-about#259] [Q&A]
Hi,

We use languages that are based on column positioning, in particular on HLASM and JCL languages, but also for COBOL.

I wish to develop snippets that insert text on precise column positions, but that does not seem planned by the mechanisms of the snippets. There is no tab tag mechanism.

It is possible to insert a `\t` tab in the text of the snippet, but this does not guarantee the position of the text to follow because the position of the tab depends on the current position and the options of the `setings.json` and not for the current snippet context.

For those who know, I would like to have the same mechanism as that existing in the ISPF Skeletons with a declaration of tabs by the `)TAB` command and the installation of a tab jump by a character declared by the `)DEFAULT` command.

For example, on the COBOL language, I would like to feed the identification area in column 73, or align my PICTURE clause in column 40.

Do you have a solution to offer?

[Snippet syntax](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_snippet-syntax)

**Comment by phaumer** [comment]
Looking for input to this as well. The only way we could think of is to provide a complete custom insert snippet implementation and not the default VS Code one as the VS Code API does not give us any APIs to customize and do things like controlling the cursor location for their snippet gallery.  Our new COBOL formatter was one ideas for how to improve the situation, but the formatter also cannot do its job if there are too many errors due to invalid column usage.
